File: Intersection_Array/intersection.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int[] Intersection(int[] nums1, int[] nums2)
    {
        HashSet<int> result = new HashSet<int>();
        Array.Sort(nums1);
        Array.Sort(nums2);

        int i = 0,
            j = 0;

        while (i < nums1.Length && j < nums2.Length)
        {
            if (nums1[i] == nums2[j])
            {
                result.Add(nums1[i]);
                i++;
                j++;
            }
            else if (nums1[i] > nums2[j])
            {
                j++;
            }
            else
            {
                i++;
            }
        }

        return result.ToArray();
    }
}

File: Intersection_Array/intersection.py
from typing import List, Dict


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1 = sorted(nums1)
        nums2 = sorted(nums2)
        intersect: List[int] = []
        check_table: Dict[int, bool] = {}

        i, j = 0, 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] == nums2[j]:
                if nums1[i] not in check_table:
                    check_table[nums1[i]] = True
                    intersect.append(nums1[i])

                i += 1
                j += 1

            elif nums1[i] > nums2[j]:
                j += 1
            else:
                i += 1

        return intersect
